diff --git a/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_1GB.dts b/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_1GB.dts
index f76b8d38f..e536be13c 100644
--- a/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_1GB.dts
+++ b/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_1GB.dts
@@ -319,31 +319,32 @@
 };
 
 &cpsw3g {
-    pinctrl-names = "default";
-    pinctrl-0 = <&rgmii1_pins_default>;
-    status = "okay";
-};
-
-&cpsw3g_mdio {
-        status = "okay";
 	pinctrl-names = "default";
-	pinctrl-0 = <&mdio1_pins_default>;
-	
-	cpsw3g_phy0: ethernet-phy@6 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <6>;
-	};
+	pinctrl-0 = <&rgmii1_pins_default>;
+		    
 };
 
 &cpsw_port1 {
-	phy-mode = "rgmii-id";
+	phy-mode = "rgmii-rxid";
 	phy-handle = <&cpsw3g_phy0>;
-	status = "okay";
 };
 
 /* port 2 not in use */
 /delete-node/ &cpsw_port2;
 
+&cpsw3g_mdio {
+	pinctrl-names = "default";
+	pinctrl-0 = <&mdio1_pins_default>;
+	status = "okay";
+
+	cpsw3g_phy0: ethernet-phy@6 {
+		reg = <6>;
+		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_00_NS>;
+		ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
+		ti,min-output-impedance;
+	};
+};
+
 &main_uart0 {
 	pinctrl-0 = <&uart0_pins_default>;
 	pinctrl-1 = <&uart0_pins_wakeup>;
diff --git a/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_256MB.dts b/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_256MB.dts
index 5215ed54d..22e9d58fc 100644
--- a/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_256MB.dts
+++ b/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_256MB.dts
@@ -325,20 +325,21 @@
 };
 
 &cpsw3g_mdio {
-        status = "okay";
 	pinctrl-names = "default";
 	pinctrl-0 = <&mdio1_pins_default>;
-	
+	status = "okay";
+
 	cpsw3g_phy0: ethernet-phy@6 {
-		compatible = "ethernet-phy-ieee802.3-c22";
 		reg = <6>;
+		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_00_NS>;
+		ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
+		ti,min-output-impedance;
 	};
 };
 
 &cpsw_port1 {
-	phy-mode = "rgmii-id";
+	phy-mode = "rgmii-rxid";
 	phy-handle = <&cpsw3g_phy0>;
-	status = "okay";
 };
 
 /* port 2 not in use */
diff --git a/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_2GB.dts b/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_2GB.dts
index 92c9d9148..7b5c6e3fd 100644
--- a/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_2GB.dts
+++ b/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_2GB.dts
@@ -325,20 +325,21 @@
 };
 
 &cpsw3g_mdio {
-        status = "okay";
 	pinctrl-names = "default";
 	pinctrl-0 = <&mdio1_pins_default>;
-	
+	status = "okay";
+
 	cpsw3g_phy0: ethernet-phy@6 {
-		compatible = "ethernet-phy-ieee802.3-c22";
 		reg = <6>;
+		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_00_NS>;
+		ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
+		ti,min-output-impedance;
 	};
 };
 
 &cpsw_port1 {
-	phy-mode = "rgmii-id";
+	phy-mode = "rgmii-rxid";
 	phy-handle = <&cpsw3g_phy0>;
-	status = "okay";
 };
 
 /* port 2 not in use */
diff --git a/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_512MB.dts b/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_512MB.dts
index febf41930..3fe1882eb 100644
--- a/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_512MB.dts
+++ b/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_512MB.dts
@@ -325,20 +325,21 @@
 };
 
 &cpsw3g_mdio {
-        status = "okay";
 	pinctrl-names = "default";
 	pinctrl-0 = <&mdio1_pins_default>;
-	
+	status = "okay";
+
 	cpsw3g_phy0: ethernet-phy@6 {
-		compatible = "ethernet-phy-ieee802.3-c22";
 		reg = <6>;
+		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_00_NS>;
+		ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
+		ti,min-output-impedance;
 	};
 };
 
 &cpsw_port1 {
-	phy-mode = "rgmii-id";
+	phy-mode = "rgmii-rxid";
 	phy-handle = <&cpsw3g_phy0>;
-	status = "okay";
 };
 
 /* port 2 not in use */
diff --git a/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_TEST.dts b/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_TEST.dts
index 9c5bc7266..661a55bd1 100644
--- a/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_TEST.dts
+++ b/arch/arm64/boot/dts/ti/k3-am62l-calixto-stamp_TEST.dts
@@ -310,20 +310,21 @@
 };
 
 &cpsw3g_mdio {
-        status = "okay";
 	pinctrl-names = "default";
 	pinctrl-0 = <&mdio1_pins_default>;
-	
+	status = "okay";
+
 	cpsw3g_phy0: ethernet-phy@7 {
-		compatible = "ethernet-phy-ieee802.3-c22";
 		reg = <7>;
+		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_00_NS>;
+		ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
+		ti,min-output-impedance;
 	};
 };
 
 &cpsw_port1 {
-	phy-mode = "rgmii-id";
+	phy-mode = "rgmii-rxid";
 	phy-handle = <&cpsw3g_phy0>;
-	status = "okay";
 };
 
 /* port 2 not in use */
